### ISSUE TITLE
Client CIO "close" doesn't work

### ISSUE THREAD

**Issue by RafalRybowski** [ktorio/ktor#900]
### Ktor Version
1.1.1
### Ktor Engine Used(client or server and name)
CIO - client
## JVM Version, Operating System and Relevant Context
Windows JVM 1.8
### Feedback
```
close(CloseReason(CloseReason.Codes.NORMAL, "Client said BYE")) for me and server five that message
after some of time because server still have connect with client which doesn't exist
Exception in thread "nettyWorkerPool-3-2" io.ktor.util.cio.ChannelWriteException: Cannot write to a channel
	at io.ktor.server.netty.cio.NettyResponsePipeline.processCallFailed(NettyResponsePipeline.kt:140)
	at io.ktor.server.netty.cio.NettyResponsePipeline.access$processCallFailed(NettyResponsePipeline.kt:22)
	at io.ktor.server.netty.cio.NettyResponsePipeline.processJobs(NettyResponsePipeline.kt:443)
	at io.ktor.server.netty.cio.NettyResponsePipeline$processJobs$1.invokeSuspend(NettyResponsePipeline.kt)
	at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:32)
	at kotlinx.coroutines.DispatchedTask.run(Dispatched.kt:233)
	at io.netty.util.concurrent.AbstractEventExecutor.safeExecute(AbstractEventExecutor.java:163)
	at io.netty.util.concurrent.SingleThreadEventExecutor.runAllTasks(SingleThreadEventExecutor.java:404)
	at io.netty.channel.nio.NioEventLoop.run(NioEventLoop.java:463)
	at io.netty.util.concurrent.SingleThreadEventExecutor$5.run(SingleThreadEventExecutor.java:884)
	at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30)
	at java.lang.Thread.run(Thread.java:748)
Caused by: java.nio.channels.ClosedChannelException
	at io.netty.channel.AbstractChannel$AbstractUnsafe.write(...)(Unknown Source)
```

**Comment by e5l** [comment]
Hi @RafalRybowski. Could you provide the code sample? 
The exception looks like from netty server.

**Comment by RafalRybowski** [comment]
Server side:
`override fun Application.main(){
        install(DefaultHeaders)
        install(WebSockets){
            pingPeriod = Duration.ofSeconds(60)
            timeout = Duration.ofSeconds(15)
            masking = false
            maxFrameSize = Long.MAX_VALUE
        }

        install(Sessions){
            cookie<RPCSession>("SESSION")
        }

        intercept(ApplicationCallPipeline.Features){
            if(call.sessions.get<RPCSession>() == null){
               call.sessions.set(RPCSession(generateNonce()))
            }
        }
        routing{
            webSocket("/RPC"){
                println("onConnect")
                val session = call.sessions.get<RPCSession>()

                if(session == null){
                    close(CloseReason(CloseReason.Codes.VIOLATED_POLICY, "No session"))
                    return@webSocket
                }

                server.memberJoin(session.id, this)
                try{
                    incoming.consumeEach { frame ->
                        if(frame is Frame.Text){
                            println("onMessage")
                            outgoing.send(element = Frame.Text(receivedMessage(session.id, frame.readText())))
                        }
                    }
                } catch (e: ClosedReceiveChannelException) {
                    println("onClose ${closeReason.await()}")
                } catch (e: Throwable) {
                    println("onError ${closeReason.await()}")
                } finally {
                    println("onClose ${closeReason.await()}")
                    server.memberLeft(session.id, this)
                }
            }
        }
    }`
Client side:
```suspend fun connect() = GlobalScope.async {
        if (host == "" && port == 0 && path == "") throw Throwable("Cannot connect with server")
        client!!.ws(host = host, port = port, path = path) {
            while (true) {
                if (close) close(CloseReason(CloseReason.Codes.NORMAL, "Client said BYE"))
                if (countTask > 0) {
                    val frame = incoming.receive()
                    when (frame) {
                        is Frame.Text -> {
                            synchronized(this) {
                                replied.add(frame.readText())
                            }
                            countTask--
                        }
                    }
                }
                if (messageList.isNotEmpty()) {
                    outgoing.send(Frame.Text(messageList[0]))
                    synchronized(this) {
                        messageList.removeAt(0)
                    }
                    countTask++
                }
            }
        }
    }```

**Comment by e5l** [comment]
Hi @RafalRybowski, it should be fixed in `1.2.0` could you recheck it?

**Comment by e5l** [comment]
Can't reproduce. Please reopen if the problem is still open.
